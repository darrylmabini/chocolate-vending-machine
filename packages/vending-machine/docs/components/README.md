React components under components directory are resusable and stateless.
