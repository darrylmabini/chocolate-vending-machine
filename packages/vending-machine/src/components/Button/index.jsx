import React from 'react';
import useCss from 'hooks/useCss';
import Styles from './Button.module.scss';

/**
 * Component to render the Button.
 */
export const Button = ({ color, children, ...restProps }) => {
  const css = useCss();
  const [colorStyle, setColorStyle] = React.useState(Styles.primary);

  React.useEffect(() => {
    switch (color) {
      case 'green':
        setColorStyle(Styles.green);
        break;

      case 'red':
        setColorStyle(Styles.red);
        break;

      case 'gray':
        setColorStyle(Styles.gray);
        break;

      default:
        setColorStyle(colorStyle);
    }
  }, [color]);

  return (
    <button className={css(Styles.button, colorStyle)} {...restProps}>
      {children}
    </button>
  );
};

Button.defaultProps = {
  color: 'primary',
};
