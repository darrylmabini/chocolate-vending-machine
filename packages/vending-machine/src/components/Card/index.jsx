import React from 'react';
import Styles from './Card.module.scss';

/**
 * Component to render the Card.
 */
export const Card = ({ children }) => <div className={Styles.card}>{children}</div>;

export const CardHeader = ({ children }) => <div className={Styles.header}>{children}</div>;

export const CardBody = ({ children }) => <div className={Styles.body}>{children}</div>;

export const CardFooter = ({ children }) => <div className={Styles.footer}>{children}</div>;
