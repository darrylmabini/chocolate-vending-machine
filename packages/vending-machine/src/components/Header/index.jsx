import React from 'react';
import Styles from './Header.module.scss';

/**
 * Component to render the Header.
 */
export const Header = ({ logo, balance }) => (
  <header className={Styles.header}>
    <div>{logo}</div>
    <div>Balance: {balance}</div>
  </header>
);

Header.defaultProps = {
  balance: '0',
  logo: null,
};
