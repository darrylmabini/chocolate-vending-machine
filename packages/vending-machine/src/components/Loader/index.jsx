import React from 'react';
import PropTypes from 'prop-types';
import useCss from 'hooks/useCss';
import Styles from './Loader.module.scss';

/**
 * Component to display a Loader.
 */
export const Loader = ({ loadingText, color }) => {
  const css = useCss();
  const [colorStyle, setColorStyle] = React.useState(Styles.violet);

  React.useEffect(() => {
    switch (color) {
      case 'white':
        setColorStyle(Styles.white);
        break;

      default:
        setColorStyle(colorStyle);
    }
  }, [color]);

  return (
    <React.Fragment>
      <div className={Styles.wrapper}>
        <div className={css(Styles.dot, colorStyle)}></div>
        <div className={css(Styles.dot, colorStyle)}></div>
        <div className={css(Styles.dot, colorStyle)}></div>
      </div>
      {loadingText && <div className={Styles.text}>{loadingText}</div>}
    </React.Fragment>
  );
};

Loader.defaultProps = {
  loadingText: null,
};

Loader.propTypes = {
  /** The loading text to display */
  loadingText: PropTypes.string,
};
