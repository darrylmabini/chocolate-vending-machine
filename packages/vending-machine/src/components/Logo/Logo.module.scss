.container {
  display: flex;
  justify-content: center;
  align-items: center;
}

.logo {
  max-height: 50px;
  max-width: 100%;
  display: block;
}

.title {
  float: left;
  margin-left: 7px;
  font-size: 1.4rem;
}
