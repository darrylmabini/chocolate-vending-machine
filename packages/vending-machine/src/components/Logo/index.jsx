import React from 'react';
import Styles from './Logo.module.scss';

/**
 * Component to display the Logo.
 */
export const Logo = ({ title, ...restProps }) => (
  <div className={Styles.container}>
    <img className={Styles.logo} {...restProps} />
    {title && <div className={Styles.title}>{title}</div>}
  </div>
);

Logo.defaultProps = {
  title: null,
  src: null,
  alt: null,
};
