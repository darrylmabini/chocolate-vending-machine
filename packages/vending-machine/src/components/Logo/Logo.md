Logo:

```jsx
import React from 'react';
import { Logo } from 'components/Logo';

<Logo />;
```
