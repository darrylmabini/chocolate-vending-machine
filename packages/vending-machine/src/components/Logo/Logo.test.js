import React from 'react';
import { render } from '@testing-library/react';
import { Logo } from 'components/Logo';

test('should renders the Logo component', () => {
  const { asFragment } = render(<Logo />);
  expect(asFragment).toMatchSnapshot();
});
