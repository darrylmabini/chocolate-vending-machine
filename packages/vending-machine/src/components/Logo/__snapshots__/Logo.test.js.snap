// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders the Logo component 1`] = `[Function]`;
